src/app/calculation/scores.py

```diff
@@ -1,12 +1,18 @@
+import pandas as pd
+from sqlalchemy import create_engine
+
 def calculate_scores_legs(leg, cur, conn):
     # annoying cleaning of input strings as sql cant handle some operators (:, $, etc)
     leg['infra_type'] = leg['infra_type'].replace("~'^parking:.*$'~'.'", "parking")
     leg['infra_type'] = leg['infra_type'].replace(":", "")
 
-    if leg['c_count'] > 0:
+    if leg['count'] > 0:
         leg['a_score'] = min(leg['a_count'] / leg['count'], 1)
         leg['c_score'] = min(leg['c_count'] / leg['count'], 1)
-        leg['p_score'] = leg['c_count'] / (leg['c_count'] + leg['a_count'])
+
+        leg['p_score'] = 0.5    # standard value when we have to avoid division by 0

```

Comment:
Maybe not calculating any scores for legs where this happens is a better idea. This does not impact the results we are working with, but looks wrong in code.

Reply:
ok. now only calculates score if leg[a_count] + leg[c_count] > 0. In this case it can calculate all scores normally

Reply:
changed to leg['c_count'] > 0 again... ist vielleicht doch am einfachsten